contracts/cw-ics721-bridge/src/ibc.rs

```diff
@@ -0,0 +1,275 @@
+use cosmwasm_schema::cw_serde;
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    from_binary, to_binary, DepsMut, Env, IbcBasicResponse, IbcChannelCloseMsg,
+    IbcChannelConnectMsg, IbcChannelOpenMsg, IbcPacket, IbcPacketAckMsg, IbcPacketReceiveMsg,
+    IbcPacketTimeoutMsg, IbcReceiveResponse, Reply, Response, StdResult, SubMsgResult, WasmMsg,
+};
+use cw_utils::parse_reply_instantiate_data;
+
+use crate::{
+    error::Never,
+    ibc_helpers::{ack_fail, ack_success, try_get_ack_error, validate_order_and_version},
+    ibc_packet_receive::do_ibc_packet_receive,
+    state::{
+        CLASS_ID_TO_NFT_CONTRACT, INCOMING_CLASS_TOKEN_TO_CHANNEL, NFT_CONTRACT_TO_CLASS_ID,
+        OUTGOING_CLASS_TOKEN_TO_CHANNEL, PROXY,
+    },
+    ContractError,
+};
+
+/// Submessage reply ID used for instantiating cw721 contracts.
+pub(crate) const INSTANTIATE_CW721_REPLY_ID: u64 = 0;
+/// Submessage reply ID used for instantiating the proxy contract.
+pub(crate) const INSTANTIATE_PROXY_REPLY_ID: u64 = 1;
+/// Submessages dispatched with this reply ID will set the ack on the
+/// response depending on if the submessage execution succeded or
+/// failed.
+pub(crate) const ACK_AND_DO_NOTHING: u64 = 2;
+/// The IBC version this contract expects to communicate with.
+pub const IBC_VERSION: &str = "ics721-1";
```

Comment:
Double check if this is right. I thought `ics721-1` is an update that includes a generic data field in the ics721 data packet.

Reply:
spec uses `ics721-1`:

https://github.com/cosmos/ibc/tree/main/spec/app/ics-721-nft-transfer#routing-module-callbacks